File: daraja/src/main/java/com/androidstudy/daraja/okhttp/AuthInterceptor.java
package com.androidstudy.daraja.okhttp;

import android.support.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthInterceptor implements Interceptor {

    private final String authToken;

    public AuthInterceptor(String authToken) {
        this.authToken = authToken;
    }

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request request = chain.request().newBuilder()
                .addHeader("Authorization", "Bearer " + authToken)
                .build();
        return chain.proceed(request);
    }
}

File: daraja/src/main/java/com/androidstudy/daraja/okhttp/AuthInterceptor.kt
package com.androidstudy.daraja.okhttp

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class AuthInterceptor(private  var authToken: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request().newBuilder()
                .addHeader("Authorization", "Bearer $authToken")
                .build()
        return chain.proceed(request)
    }

}